src/lxcfs.c

```diff
@@ -149,31 +151,41 @@ static void do_reload(void)
 good:
 	if (loadavg_pid > 0)
 		start_loadavg();
-
-	if (need_reload)
-		lxcfs_info("Reloaded LXCFS");
-	need_reload = 0;
 }
 
 static void up_users(void)
 {
-	users_lock();
-	if (users_count == 0 && need_reload)
-		do_reload();
-	users_count++;
-	users_unlock();
+	if (atomic_load_acquire(&need_reload)) {
```

Comment:
Doing this before the `increment` can be dangerous, though.
Assume thread 1 goes to the *top* of the `else` branch before incrementing, `user_count` is still `0`.
Then `need_reload` gets set and another thread goes into this branch right up to the `do_reload()`, and right in the middle of it, the scheduler switches back to thread 1 which bumps the counter and continues towards a crash.

Reply:
Yeah, I already had deadlocks anyway.